src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -563,27 +537,252 @@ internal SafeFileHandle OpenPort(string portName)
                 _handle = null;
                 throw;
             }
+
+            _processReadDelegate = ProcessRead;
+            _processWriteDelegate = ProcessWrite;
+            _ioLoop = Task.Factory.StartNew(
+                IOLoop,
+                CancellationToken.None,
+                TaskCreationOptions.LongRunning,
+                TaskScheduler.Default);
         }
 
-        ~SerialStream()
+        private void FinishPendingIORequests()
         {
-            Dispose(false);
+            while (_readQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
+
+            while (_writeQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
         }
 
         protected override void Dispose(bool disposing)
         {
-            // Signal the other side that we're closing.  Should do regardless of whether we've called
-            // Close() or not Dispose()
-            if (_handle != null && !_handle.IsInvalid)
+            if (disposing)
             {
-                   Interop.Sys.Shutdown(_handle, SocketShutdown.Both);
-                   _handle.Close();
+                _ioLoopFinished = true;
+                _ioLoop?.Wait();
```

Comment:
Nit: `.Wait()` => `.GetAwaiter().GetResult();`... that way if this does fail with an exception, that exception will be propagated directly rather than being wrapped in an AggregateException.